source/slang/hlsl.meta.slang

```diff
@@ -11580,9 +11580,12 @@ void SetMeshOutputCounts(uint vertexCount, uint primitiveCount)
 //
 // This function doesn't return.
 //
+// This function cannot be inlined due to a legalization pass happening mid-way through processing
+// and later more processing happening to the function which requires eventual inlining.
 [KnownBuiltin("DispatchMesh")]
 [require(glsl_hlsl_spirv, meshshading)]
-void DispatchMesh<P>(uint threadGroupCountX, uint threadGroupCountY, uint threadGroupCountZ, __constref P meshPayload)
+[noRefInline]
```

Comment:
This feels like a really hacky way to get things working, particularly when this is the only case that we are trying to workaround here. The glsl legalization logic probably needs a revisit so that it is more conformant with how rest of the compiler is written. We should adopt something similar to your ray payload technique here to replace `meshPayload` with a global index.

Reply:
do we merge this pr, make a seperate issue, and redesign it later?

Reply:
Yes, let's merge the pr first.